src/coreclr/tools/Common/TypeSystem/Ecma/EcmaMethod.cs

```diff
@@ -599,5 +599,7 @@ private MarshalAsDescriptor GetMarshalAsDescriptor(Parameter parameter)
         public override EcmaMethod GetMethodDefinition() => this;
 
         public override EcmaMethod GetTypicalMethodDefinition() => this;
+
+        public override MethodDesc InstantiateSignature(Instantiation typeInstantiation, Instantiation methodInstantiation) => this;
```

Comment:
Use covariant return types to make it easier to reduce casting of the return type in callers?

```suggestion
        public override EcmaMethod InstantiateSignature(Instantiation typeInstantiation, Instantiation methodInstantiation) => this;
```

Reply:
And do the same in AsyncMethodVariant.cs too?